File: churn-demo/old/account.py
class Account:
    def __init__(self, owner, balance):
        self.owner = owner
        self.balance = balance
        self.history = []

    def deposit(self, amount):
        self.balance += amount
        self.history.append("deposit")

    def withdraw(self, amount):
        self.balance -= amount
        self.history.append("withdraw")

    def apply_fee(self, fee):
        self.balance -= fee
        self.history.append("fee")

    def get_balance(self):
        return self.balance

    def is_overdrawn(self):
        return self.balance < 0

    def history_count(self):
        return len(self.history)

    def print_statement(self):
        for e in self.history:
            print(e)

File: churn-demo/old/Account.java
import java.util.ArrayList;
import java.util.List;

public class Account {
    private String owner;
    private double balance;
    private List<String> history = new ArrayList<>();

    public Account(String owner, double balance) {
        this.owner = owner;
        this.balance = balance;
    }

    public void deposit(double amount) {
        balance += amount;
        history.add("deposit");
    }

    public void withdraw(double amount) {
        balance -= amount;
        history.add("withdraw");
    }

    public void applyFee(double fee) {
        balance -= fee;
        history.add("fee");
    }

    public double getBalance() {
        return balance;
    }

    public boolean isOverdrawn() {
        return balance < 0;
    }

    public int historyCount() {
        return history.size();
    }

    public void printStatement() {
        for (String e : history) {
            System.out.println(e);
        }
    }
}
